packages/bot-web-ui/src/components/chart/chart.jsx

```diff
@@ -13,77 +14,96 @@ import '../../assets/sass/chart.scss';
 // import { symbolChange }      from '../../SmartChart/Helpers/symbol';
 // import AllMarkers            from '../../SmartChart/Components/all-markers.jsx';
 
-class Chart extends React.Component {
-    chartTopWidgets = () => <ChartTitle onChange={this.props.onSymbolChange} />;
+const ChartTopWidgets = ({ onChange }) => <ChartTitle onChange={onChange} />;
 
-    chartToolbarWidget = () => (
-        <ToolbarWidgets updateChartType={this.props.updateChartType} updateGranularity={this.props.updateGranularity} />
-    );
+const ChartToolbarWidget = ({ updateChartType, updateGranularity }) => (
+    <ToolbarWidgets updateChartType={updateChartType} updateGranularity={updateGranularity} />
+);
 
+const Chart = ({
+    granularity,
+    chart_type,
+    is_mobile,
+    is_socket_opened,
+    onSymbolChange,
+    updateChartType,
+    updateGranularity,
+    show_digits_stats,
+    // should_refresh,
+    // resetRefresh,
+    wsForget,
+    wsForgetStream,
+    wsSendRequest,
+    wsSubscribe,
+    setChartStatus,
+    settings,
+    symbol,
+}) => {
     // bottomWidgets = ({ digits, tick }) => (
     //     <ChartBottomWidgets digits={digits} tick={tick} />
     // );
 
-    render() {
-        const {
-            show_digits_stats,
-            main_barrier,
-            // should_refresh,
-            // resetRefresh,
-            chart_type,
-            is_mobile,
-            granularity,
-            wsSendRequest,
-            wsForget,
-            wsForgetStream,
-            wsSubscribe,
-            settings,
-            symbol,
-            is_socket_opened,
-        } = this.props;
+    const barriers = [];
```

Comment:
We removed `main_barrier ? [main_barrier] : [];`, is this intentional?

Reply:
Yes, this prop never update and main_barrier is always null